diff-containers/README.md

```diff
@@ -1 +1,238 @@
-# diff-containers
\ No newline at end of file
+# diff-containers
+
+This package implements difference datatypes for container types. Currently,
+there is only one such implemented difference datatype for `Data.Map.Strict`.
+
+## Differences on `Data.Map`s
+
+A difference datatype for maps is defined and implemented in the
+`Data.Map.Diff.Strict` module. A key property of our definition of diffs is that
+diffs can not only be combined/summed, they can also be inverted and summed
+together, which facilitates *subtraction* of diffs. Our definition of diffs is
+not only a `Semigroup` and a `Monoid`, it is also a `Group`.
```

Comment:
I wonder if it'd make sense to include links to each of these algebraic data structures (maybe to Wikipedia).

Reply:
Done